udp-reassembly-offload.md

```diff
@@ -27,6 +27,25 @@ This section describes necessary updates in the Windows network stack to support
 
 The NDIS interface for URO is used for communication between TCPIP and the NDIS miniport driver.
 
+## Rules
+
+URO can only be attempted on a batch of packets that meet all the following criteria:
+
+- Unicast only (no broadcast or multicast)
+- 5-tuple matches
+- Payload length is identical for all datagrams
+- ECN, DF bits must match on all packets (IPv4)
+- NextHeader must be UDP (IPv6)
+- The total length of the Single Coalesced Unit (SCU) must not exceed IP max length
+- No more than 255 NBLs per SCU
```

Comment:
We are talking NBLs here. Is this description for software URO?

Reply:
Based on the software URO implementation, so this requirement is mutable.

Reply:
Right, this doesn't make much sense because each URO packet is by definition a single NBL.